### PR TITLE
fix: INTMDB-1158 search index analyzers

### PR DESCRIPTION
**Pull request by Adelmar92**
## Proposed changes
Fixed search analysers for search index analyzer structure

## Test

Ive created a stack with an Analyzer structure:
image

i finally validate in the Atlas UI that the analyzer is created correctly: 
image

https://jira.mongodb.org/browse/INTMDB-1158

_Jira ticket:_ CLOUDP-#

Please include a summary of the fix/feature/change, including any relevant motivation and context.

Link to any related issue(s): 

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
  expected)
- [ ] This change requires a documentation update

## Manual QA performed:

- [ ] cfn invoke for each of CRUDL/cfn test
- [ ] Updated resource in  [example](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples)
- [ ] Published to AWS private registry
- [ ] Used the template in [example](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples) to create and update a stack in AWS
- [ ] Deleted stack to ensure resources are deleted
- [ ] Created multiple resources in same stack
- [ ] Validated in Atlas UI
- [ ] Included screenshots

## Required Checklist:

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run `make fmt` and formatted my code
- [ ] For CFN Resources: I have released by changes in the private registry and proved by change
  works in Atlas

### CODE REVIEW

File: cfn-resources/search-index/cmd/resource/resource.go [hunk 1]
```diff
@@ -152,43 +158,41 @@ func newSearchIndex(currentModel *Model) (*admin.ClusterSearchIndex, error) {
 }
 
 // convertToAnySlice function converts a slice of map[string]interface{} to a slice of interface{}
-func convertToAnySlice(input []map[string]interface{}) []interface{} {
-	result := make([]interface{}, len(input))
-	for i, v := range input {
-		result[i] = v
+func convertToAnySlice(input []string) ([]interface{}, error) {
```
Comment (marcosuma) [inline cfn-resources/search-index/cmd/resource/resource.go:161]: please unit test this

File: cfn-resources/search-index/cmd/resource/resource.go [hunk 2]
```diff
@@ -152,43 +158,41 @@ func newSearchIndex(currentModel *Model) (*admin.ClusterSearchIndex, error) {
...
-	}
-	if v, ok := tokenizer["minGram"]; ok {
-		result.MinGram = admin.PtrInt(cast.ToInt(v))
-	}
-	if v, ok := tokenizer["type"]; ok {
-		result.Type = admin.PtrString(cast.ToString(v))
-	}
-	if v, ok := tokenizer["group"]; ok {
-		result.Group = admin.PtrInt(cast.ToInt(v))
+func newTokenizerModel(tokenizer *ApiAtlasFTSAnalyzersTokenizer) admin.ApiAtlasFTSAnalyzersTokenizer {
```
Comment (marcosuma) [inline cfn-resources/search-index/cmd/resource/resource.go:176]: please unit test this

File: cfn-resources/search-index/cmd/resource/resource.go [hunk 3]
```diff
@@ -198,23 +202,16 @@ func newMappings(currentModel *Model) (*admin.ApiAtlasFTSMappings, error) {
 	}, nil
 }
 
-func newMappingsFields(fields []string) (map[string]interface{}, error) {
-	if len(fields) == 0 {
+func convertStringToInterface(fields *string) (map[string]interface{}, error) {
```
Comment (marcosuma) [inline cfn-resources/search-index/cmd/resource/resource.go:205]: please unit test this

File: cfn-resources/search-index/cmd/resource/resource.go (hunk 2 above)
Comment (Adelmar92) [reply cfn-resources/search-index/cmd/resource/resource.go:176]: i dont think we need unit testing for this one, is just a straight conversion

**Review by lantoli — APPROVED** [review]

**Review by lantoli — APPROVED** [review]

**Review by lantoli — APPROVED** [review]

**Pull request merged**